tiledb/sm/array/array.h

```diff
@@ -205,19 +206,26 @@ class Array {
   Status reopen();
 
   /**
-   * Re-opens the array at a specific timestamp.
+   * Re-opens the array at a specific timestamp. If `timestamp_start` is
+   * provided, this will re-open the array between the two timestamps.
    *
    * @note Applicable only for reads, it errors if the array was opened
    *     for writes.
    */
-  Status reopen(uint64_t timestamp);
+  Status reopen(uint64_t timestamp_end, uint64_t timestamp_start = 0);
```

Comment:
I see the need to put `timestamp_start` as the last argument to use a default value, but in this particular case I advise against as it is very error prone. Let's use start first and then end.